frontend/src/app/projects/models/diagrams/model-diagram-dialog/model-diagram-dialog.component.ts

```diff
@@ -25,15 +27,20 @@ import {
   DiagramMetadata,
   ModelDiagramService,
 } from 'src/app/projects/models/diagrams/service/model-diagram.service';
+import { environment } from 'src/environments/environment';
 
 @Component({
   selector: 'app-model-diagram-dialog',
   templateUrl: './model-diagram-dialog.component.html',
   styleUrls: ['./model-diagram-dialog.component.css'],
 })
-export class ModelDiagramDialogComponent {
+export class ModelDiagramDialogComponent implements OnInit {
   diagramMetadata?: DiagramCacheMetadata;
   diagrams: Diagrams = {};
+  cachedUsername: string | null = null;
```

Comment:
I would simply name this `username` instead of `cachedUsername`. In addition, I would also use `undefined` as default value (we typically use `undefined` for the case where data were not yet loaded but should be there, e.g., when initialising the value via an api call). So it can look like this: `username?: string` (short for `username: string | undefined = undefined`.

The actual initialisation can then be done inside of the constructor as follows:

```typescript
this.userService
      .getCurrentUser()
      .subscribe((user) => (this.username = user.name));
```